src/main/java/org/apache/commons/lang3/StringUtils.java

```diff
@@ -3277,6 +3277,164 @@ public static String substringBetween(final String str, final String open, final
         return list.toArray(new String[list.size()]);
     }
 
+    /**
+     * <p>Split a String into an array, using an array of fixed string lengths.</p>
+     *
+     * <p>If not null String input, the returned array size is same as the input lengths array.</p>
+     *
+     * <p>A null input String returns {@code null}.
+     * A {@code null} or empty input lengths array returns an empty array.
+     * A {@code 0} in the input lengths array results in en empty string.</p>
+     *
+     * <p>Extra characters are ignored (ie String length greater than sum of split lengths).
+     * All empty substrings other than zero length requested, are returned {@code null}.</p>
+     *
+     * <pre>
+     * StringUtils.splitByLength(null, *)      = null
+     * StringUtils.splitByLength("abc")        = []
+     * StringUtils.splitByLength("abc", null)  = []
+     * StringUtils.splitByLength("abc", [])    = []
+     * StringUtils.splitByLength("", 2, 4, 1)  = [null, null, null]
+     *
+     * StringUtils.splitByLength("abcdefg", 2, 4, 1)     = ["ab", "cdef", "g"]
+     * StringUtils.splitByLength("abcdefghij", 2, 4, 1)  = ["ab", "cdef", "g"]
+     * StringUtils.splitByLength("abcdefg", 2, 4, 5)     = ["ab", "cdef", "g"]
+     * StringUtils.splitByLength("abcdef", 2, 4, 1)      = ["ab", "cdef", null]
```

Comment:
Good point.
My idea was to indicate that there is no more characters to extract the explicitly requested column length text.
But in the other hand, this can cause `NullPointerException` if the returned array is used without check...

Why I used this approach : with `null` values, in case of hard coded lengths, we can simply check the returned array with a _for each_ loop, even later in other piece of code :
```java
String[] cols = StringUtils.splitByLength(input, 2, 3, 1);
// ...
for (String col : cols) {
    if (col == null) {
        break;
    }
    // Do something
}
```

Without `null` values, we must have a lengths array reference :
```java
int[] LENGTHS = { 2, 3, 0, 1 };
String[] cols = StringUtils.splitByLength(input, LENGTHS);
int index = 0;
for (String col : cols) {
    if (col.length() == 0 && LENGTHS[index] > 0) {
        break;
    }
    index++;
    // Do something
}
```

Of course, we can also check the input string length before calling `StringUtils.splitByLength`, but we have to get the lengths sum. And what about this case : `StringUtils.splitByLength("abcd", 1, 2, 2)` ?

I don't know which is best... What do you think ?